zipline/pipeline/factors/factor.py

```diff
@@ -1224,9 +1240,79 @@ def compute(self, today, assets, out, data):
         # MedianValue.
         median_close10 = MedianValue([USEquityPricing.close], window_length=10)
         median_low15 = MedianValue([USEquityPricing.low], window_length=15)
+
+    A CustomFactor with multiple outputs:
+
+    .. code-block:: python
+
+        class MultipleOutputs(CustomFactor):
+            inputs = [USEquityPricing.close]
+            outputs = ['alpha', 'beta']
+            window_length = N
+
+            def compute(self, today, assets, out, close):
+                computed_alpha, computed_beta = some_function(close)
+                out.alpha = computed_alpha
+                out.beta = computed_beta
+
+        # Each output is returned as its own Factor upon instantiation.
+        alpha, beta = MultipleOutputs()
     '''
     dtype = float64_dtype
 
+    def __getattr__(self, name):
+        if name in self.outputs:
+            return RecarrayFactor(factor=self, attribute=name)
+        else:
+            raise AttributeError(
+                "This factor has no output called '{}'.".format(name)
+            )
```

Comment:
This is for allowing something like:

``` python
class MyFactor(CustomFactor):
    outputs = ['alpha', 'beta']
    ...
my_factor = MyFactor
alpha, beta = my_factor.alpha, my_factor.beta
```

Should we keep both this and `__iter__` and allow users to choose how to access the outputs, or just pick one single option?